src/AdminBundle/Controller/UserController.php

```diff
@@ -106,47 +168,49 @@ public function newAction()
         return array(
             'entity' => $entity,
             'form'   => $form->createView(),
+            'errors' => null
         );
     }
 
     /**
-     * Finds and displays a User entity.
-     * @param Request $request
-     * @return array
-     * 
      * @Route("/{id}", name="admin_users_show")
      * @Method("GET")
      * @Template()
      */
-    public function showAction($id, Request $request)
+    public function showAction(User $user, Request $request)
     {
         $em = $this->getDoctrine()->getManager();
 
-        /** @var User $entity */
-        $entity = $em->getRepository('AppBundle:User')->findOneBy(['id' => $id]);
-
-        if (!$entity) {
-            $this->addFlash('danger', 'No user was found for this id.');
-            return $this->redirectToRoute('admin_users');
-        }
-
-        if (!$entity->getLocation()) {
+        if (!$user->getLocation()) {
             $this->addFlash('danger', 'User must have location.');
             return $this->redirectToRoute('admin_users');
         }
 
-        if ($entity->getLocation() && !$entity->getLocation()->getCity()) {
+        if ($user->getLocation() && !$user->getLocation()->getCity()) {
             $this->addFlash('danger', 'User must have city in location.');
             return $this->redirectToRoute('admin_users');
         }
 
-        $deleteForm = $this->createDeleteForm($id);
+        $deleteForm = $this->createDeleteForm($user->getId());
 
-        $em    = $this->get('doctrine.orm.entity_manager');
-        $dql   = "SELECT a FROM AppBundle:Project a LEFT JOIN a.voteSetting vs LEFT JOIN vs.location l WHERE l.city = :city ORDER BY a.id DESC";
+        /** @var VoteSettings[] $voteSettings */
+        $voteSettings = $this->getDoctrine()->getRepository('AppBundle:VoteSettings')->getVoteSettingByUserCity($user);
```

Comment:
`'AppBundle:VoteSettings'` to VoteSettings::class

Reply:
you wrote this line, and you can change it